site/content/en/latest/design/eg-metrics.md

```diff
@@ -0,0 +1,225 @@
+
+I propose the following:
+
+#### Metrics
+
+Metrics offer the greatest potential for providing guarantees. They often directly influence alerts and dashboards, making changes highly impactful. This contrasts with traces and logs, which are often used for ad-hoc analysis, where minor changes to information can be easily understood by a human.
+
+Moreover, there is precedent for this: [Kubernetes Metrics Lifecycle](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/#metric-lifecycle) has well-defined processes, and Envoy Gateway's dataplane (Envoy Proxy) metrics are de facto stable.
+
+Currently, all Envoy Gateway metrics lack defined stability. I suggest we categorize all existing metrics as either:
+
++ ***Deprecated***: a metric that is intended to be phased out.
++ ***Experimental***: a metric that is off by default.
++ ***Alpha***: a metric that is on by default.
+
+We should aim to promote a core set of metrics to **Stable** within a few releases.
+
+## Envoy Gateway API Types
+
+New APIs will be added to Envoy Gateway config, which are used to manage Control Plane Telemetry bootstrap configs.
+
+### EnvoyGatewayTelemetry
+
+``` go
+// EnvoyGatewayTelemetry defines telemetry configurations for envoy gateway control plane.
+// Control plane will focus on metrics observability telemetry and tracing telemetry later.
+type EnvoyGatewayTelemetry struct {
+	// Metrics defines metrics configuration for envoy gateway.
+	Metrics *EnvoyGatewayMetrics `json:"metrics,omitempty"`
+}
+```
+
+### EnvoyGatewayMetrics
+
+> Prometheus will be exposed on 0.0.0.0:19001, which is not supported to be configured yet.
+
+``` go
+// EnvoyGatewayMetrics defines control plane push/pull metrics configurations.
+type EnvoyGatewayMetrics struct {
+	// Sinks defines the metric sinks where metrics are sent to.
+	Sinks []EnvoyGatewayMetricSink `json:"sinks,omitempty"`
+	// Prometheus defines the configuration for prometheus endpoint.
+	Prometheus *EnvoyGatewayPrometheusProvider `json:"prometheus,omitempty"`
+}
+
+// EnvoyGatewayMetricSink defines control plane
+// metric sinks where metrics are sent to.
+type EnvoyGatewayMetricSink struct {
+	// Type defines the metric sink type.
+	// EG control plane currently supports OpenTelemetry.
+	// +kubebuilder:validation:Enum=OpenTelemetry
+	// +kubebuilder:default=OpenTelemetry
+	Type MetricSinkType `json:"type"`
+	// OpenTelemetry defines the configuration for OpenTelemetry sink.
+	// It's required if the sink type is OpenTelemetry.
+	OpenTelemetry *EnvoyGatewayOpenTelemetrySink `json:"openTelemetry,omitempty"`
+}
+
+type EnvoyGatewayOpenTelemetrySink struct {
+	// Host define the sink service hostname.
+	Host string `json:"host"`
+	// Protocol define the sink service protocol.
+	// +kubebuilder:validation:Enum=grpc;http
+	Protocol string `json:"protocol"`
```

Comment:
from the default value of port, I suspect default value for this field is `grpc`.

Reply:
imo its fine to start w/o a default